### PR TITLE
Ping status fix

### PR DESCRIPTION
**Pull request by aneeshkp**
passing clinetID to know which client the event to be sent to .this fix is for http only

Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>

### CODE REVIEW

File: pkg/protocol/http/http.go [hunk 1]
```diff
@@ -407,7 +401,37 @@ func (h *Server) HTTPProcessor(wg *sync.WaitGroup) {
 								OnReceiveOverrideFn: d.OnReceiveOverrideFn,
 								ProcessEventFn:      d.ProcessEventFn,
 							}
-							h.SendTo(wg, clientID, endPointURI.String(), data.Data, d.Type)
+							h.SendTo(wg, d.ClientID, *url, data.Data, d.Type)
+							log.Infof("status ping: queued event status for client %s  for resource %s", *url, d.Address)
```
Comment (jzding) [inline pkg/protocol/http/http.go:405]: pass `d.ClientID` to `client %s`.

File: pkg/protocol/http/http.go [hunk 2]
```diff
@@ -407,7 +401,37 @@ func (h *Server) HTTPProcessor(wg *sync.WaitGroup) {
...
+						if len(eventSubscribers) == 0 {
+							log.Errorf("no subscriber found for resource %s", d.Address)
+							log.Errorf("event not publsihed to empty subscribers, clients need to register %s", d.Address)
+							log.Infof("event to log %s", d.Data.String())
+							d.Status = channel.FAILED
+							localmetrics.UpdateEventCreatedCount(d.Address, localmetrics.FAILED, -1)
+							h.DataOut <- d
+						} else {
+							for clientID, endPointURI := range eventSubscribers {
+								log.Infof("event to log %s", d.Data.String())
```
Comment (jzding) [inline pkg/protocol/http/http.go:421]: debugf ? data might be big

File: pkg/channel/data.go [hunk 3]
```diff
@@ -16,21 +16,34 @@ package channel
...
-	// if not set then the data is sent to out channel and processed by side car  default method
+	// if not set then the data is sent to out channel and processed by side-car  default method
 	OnReceiveOverrideFn func(e cloudevents.Event, dataChan *DataChan) error
 	// ProcessEventFn  Optional, this allows to customize message handler thar was received at the out channel
 	ProcessEventFn func(e interface{}) error
 }
+
+// CreateCloudEvents ...
+func (d *DataChan) CreateCloudEvents(dataType string) (*cloudevents.Event, error) {
+	ce := cloudevents.NewEvent(cloudevents.VersionV03)
```
[the remark this answers is not in the document]
Comment (aneeshkp) [reply pkg/channel/data.go:42]: that is not my code. It belongs  to cloudevents library

File: pkg/protocol/http/http.go (hunk 2 above)
Comment (aneeshkp) [reply pkg/protocol/http/http.go:421]: ok, will update

File: pkg/protocol/http/http.go (hunk 1 above)
Comment (aneeshkp) [reply pkg/protocol/http/http.go]: done

**Review by jzding — APPROVED** [review]
/lgtm

**Pull request merged**